test/unit/services/validators/test-common-validator.js

```diff
@@ -0,0 +1,129 @@
+
+describe('services/validators/common-validator', function () {
+
+  describe('isNullOrUndefined', function () {
+    it('should return true if passed null', function () {
+      var result = validator.isNullOrUndefined(null)
+      expect(result).to.equal(true)
+    })
+
+    it('should return true if passed undefined', function () {
+      var result = validator.isNullOrUndefined(undefined)
+      expect(result).to.equal(true)
+    })
+
+    it('should return true if passed an empty string', function () {
+      var result = validator.isNullOrUndefined('')
+      expect(result).to.equal(true)
+    })
+
+    it('should return false if passed an object', function () {
+      var result = validator.isNullOrUndefined({})
+      expect(result).to.equal(false)
+    })
+
+    it('should return false if passed an array', function () {
+      var result = validator.isNullOrUndefined([])
+      expect(result).to.equal(false)
+    })
+
+    it('should return false if passed a non empty string', function () {
+      var result = validator.isNullOrUndefined('any string')
+      expect(result).to.equal(false)
+    })
+  })
+
+  describe('isValidDate', function () {
+    it('should return false if passed null', function () {
+      var result = validator.isValidDate(null)
+      expect(result).to.equal(false)
+    })
+
+    it('should return false if passed undefined', function () {
+      var result = validator.isValidDate(undefined)
+      expect(result).to.equal(false)
+    })
+
+    it('should return false if passed an object', function () {
+      var result = validator.isValidDate({})
+      expect(result).to.equal(false)
+    })
+
+    it('should return true if passed a valid Date object', function () {
+      var result = validator.isValidDate(dateFormatter.now())
+      expect(result).to.equal(true)
+    })
+
+  })
+
+  describe('isDateInThePast', function () {
+    const PAST_DATE = dateFormatter.now().subtract(1, 'day')
+    const FUTURE_DATE = dateFormatter.now().add(1, 'day')
+
+    it('should return false if passed null', function () {
+      var result = validator.isDateInThePast(null)
+      expect(result).to.equal(false)
+    })
+
+    it('should return false if passed undefined', function () {
+      var result = validator.isDateInThePast(undefined)
+      expect(result).to.equal(false)
+    })
+
+    it('should return false if passed an object', function () {
+      var result = validator.isDateInThePast({})
+      expect(result).to.equal(false)
+    })
+
+    it('should return true if passed a valid Date object', function () {
+      var result = validator.isDateInThePast(PAST_DATE)
+      expect(result).to.equal(true)
+    })
+
+    it('should return false if passed an invalid Date object', function () {
+      var result = validator.isDateInThePast(FUTURE_DATE)
+      expect(result).to.equal(false)
+    })
+  })
+
+  describe('isDateOlderThanMaxHistory', function () {
+    const MAX_HISTORY = 6
```

Comment:
Now that we have this in a constant we can include it so that future updates only require changing one place.